### ISSUE TITLE
Make Array.sort order arg optional to match js implementation.

### ISSUE THREAD

**Issue by jessekelly881** [Effect-TS/effect#2600]
### What is the problem this feature would solve?

`Array.prototype.sort`s compare function is optional and it automatically handles sorting number[], string[], etc. I think we should do the same with Array.sort(et al) making the Order arg optional and relying on the default whenever possible. `Array.sort()` is more convenient and would be more performant than `Array.sort(Order.number)` or `Array.sort(Order.string)`

### What is the feature you are proposing to solve the problem?

^

**Comment by mikearnaldi** [comment]
Array.sort is mutable so it can't be used directly, we'd have to clone the array leading to the same perf tradeoff

**Comment by jessekelly881** [comment]
Yeah, I mean, there is toSorted which would probably be better when there is a little more browser support (currently at 80%). But I don't think the mutability of .sort is relevant here. Copied or not it would be better to rely on the default comparison algorithm as a default (works for numbers, strings).

**Comment by mikearnaldi** [comment]
> would be more performant than `Array.sort(Order.number)` or `Array.sort(Order.string)`

was answering to:

```
would be more performant than Array.sort(Order.number) or Array.sort(Order.string)
```

**Comment by fubhy** [comment]
Closing this as implemented in the current repository. If the current API does not cover your use case, please open a new issue with the missing behavior.
